core/src/cache.rs

```diff
@@ -1068,6 +1068,18 @@ impl Cache {
         self.terms.get(&file_id).map(|TermEntry { term, .. }| term)
     }
 
+    /// Set a new value for a cached term.
+    pub fn set(&mut self, file_id: FileId, term: RichTerm, state: EntryState) {
```

Comment:
It is definitely suspicious and I've removed it.

I replaced it with the `custom_transform` function, which I'm also not completely thrilled with but I think it's better than the previous poking around in `Program`'s private fields.

Fundamentally, the problem is that we want to apply a custom transformation pass, but otherwise we want all of the behavior that `Program` provides (the input processing, the override mode, the error reporting...)